kore/src/Kore/Reachability/Prove.hs

```diff
@@ -420,6 +422,28 @@ transitionRule' claims axioms = \prim proofState ->
   where
     axiomGroups = groupSortOn Attribute.Axiom.getPriorityOfAxiom axioms
 
+withWarnings
+    :: forall m
+    .  MonadSimplify m
+    =>  CommonTransitionRule m
+    -> CommonTransitionRule m
+withWarnings rule prim claimState =
+    case prim of
+        Prim.CheckImplication  ->
+            case ClaimState.retractRewritable claimState of
+                Just claim -> do
+                    result <- checkImplication claim & Logic.lowerLogicT
```

Comment:
This should never call `checkImplication`. It may only call `rule` and inspect the inputs and outputs.